### PR TITLE
Breadcrumb link for single order edit page, and some style structuring.

### PR DESCRIPTION
**Pull request by stratease**
### 🎫 Ticket

[ET-2184]

### 🗒️ Description
Added breadcrumb for the Order Detail page, and did some css cleanup.

### 🎥 Artifacts 

image

### ✔️ Checklist
- [x] Ran `npm run changelog` to add changelog file(s).
- [x] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [x] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[ET-2184]: https://stellarwp.atlassian.net/browse/ET-2184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Admin/Singular_Order_Page.php [hunk 1]
```diff
@@ -56,6 +56,27 @@ public function register() {
...
+	 *
+	 * @param string $classes The classes string for the body attribute.
+	 *
+	 * @return mixed|string The mutated classes string.
+	 */
+	public function add_body_class( $classes ) {
+		$screen = get_current_screen();
+
+		// Check if we are on the single edit screen of the custom post type.
+		if ( $screen && $screen->post_type === Order::POSTTYPE && $screen->base === 'post' ) {
```
[the remark this answers is not in the document]
Comment (stratease) [reply src/Tickets/Commerce/Admin/Singular_Order_Page.php:75]: Didn't we get rid of these?

File: src/Tickets/Commerce/Admin/Singular_Order_Page.php [hunk 2]
```diff
@@ -56,6 +56,64 @@ public function register() {
...
+	 *
+	 * @since TBD
+	 */
+	public function render_breadcrumb_order_edit_screen_html() {
+		$url  = get_current_screen()->parent_file;
+		$text = _x( 'Orders', 'Order edit page breadcrumb link back to Orders page.', 'event-tickets' );
+
+		$html = <<<STR
+		<div class="tec-tickets-commerce-single-order--breadcrumb--order--edit">
+			<a href="$url"><span class="dashicons dashicons-arrow-left-alt2"></span> $text</a>
```
Comment (dpanta94) [inline src/Tickets/Commerce/Admin/Singular_Order_Page.php:93]: Escape vars

File: src/Tickets/Commerce/Admin/Singular_Order_Page.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (dpanta94) [reply src/Tickets/Commerce/Admin/Singular_Order_Page.php:112]: The yoda conditions ? i dont think so. We could, i think the purpose they serve is not significant.

File: tests/integration/TEC/Tickets/Commerce/Admin/__snapshots__/Singular_Order_PageTest__it_should_render_breadcrumb__0.snapshot.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+		<div class="tec-tickets-commerce-single-order--breadcrumb--order--edit">
+			<a href="edit.php?post_type=tec_tc_order"><span class="dashicons dashicons-arrow-left-alt2"></span> Orders</a>
```
Comment (dpanta94) [inline tests/integration/TEC/Tickets/Commerce/Admin/__snapshots__/Singular_Order_PageTest__it_should_render_breadcrumb__0.snapshot.html:2]: Hmm maybe we need an `admin_url()` in our code here ?

File: tests/integration/TEC/Tickets/Emails/Admin/__snapshots__/Emails_TabTest__it_should_match_stored_json_for_general_settings__1.php [hunk 4]
```diff
@@ -17,9 +17,13 @@
...
-        "html": "<h3>Sender Information<\\/h3>"
+        "html": "<div class=\\"tribe-common\\">\\n\\t<div  class=\\"tec-admin__upsell tec-admin__upsell-tec-tickets-wallet-plus\\" >\\n\\t\\t<div class=\\"tec-admin__upsell-content\\">\\n\\t\\t\\t<div class=\\"tec-admin__upsell-icon\\">\\n\\t<img\\n\\t\\tclass=\\"tec-admin__upsell-icon-image\\"\\n\\t\\tsrc=\\"http:\\/\\/wordpress.test\\/wp-content\\/plugins\\/the-events-calendar\\/common\\/src\\/resources\\/images\\/icons\\/circle-bolt.svg\\"\\n\\t\\talt=\\"The Events Calendar important notice icon\\"\\n\\t\\/>\\n<\\/div>\\t\\t\\t<div class=\\"tec-admin__upsell-text-wrap\\">\\n\\t\\t\\t\\t<span class=\\"tec-admin__upsell-text\\">\\n\\t\\t\\t\\t\\tGet additional ticketing flexibility including Apple Wallet and PDF tickets with \\t\\t\\t\\t<\\/span>\\n\\t\\t\\t\\t<a \\n\\t\\t\\t\\t\\t class=\\"tec-admin__upsell-link tec-admin__upsell-link--underlined\\"  \\n\\t\\t\\t\\t\\thref=\\"https:\\/\\/evnt.is\\/1bdz\\" \\n\\t\\t\\t\\t\\trel=\\"noopener noreferrer\\"\\n\\t\\t\\t\\t\\ttarget=\\"_blank\\"\\n\\t\\t\\t\\t>\\n\\t\\t\\t\\t\\tEvent Tickets Plus\\t\\t\\t\\t<\\/a>\\n\\t\\t\\t<\\/div>\\n\\t\\t<\\/div>\\n\\t<\\/div>\\n<\\/div>"
```
Comment (dpanta94) [inline tests/integration/TEC/Tickets/Emails/Admin/__snapshots__/Emails_TabTest__it_should_match_stored_json_for_general_settings__1.php:20]: Why is this snapshot being changed ?

File: tests/integration/Tribe/Admin/__snapshots__/OrderReportTest__test_tc_order_report_display__event with 1 pending and 1 completed order__0.snapshot.html [hunk 5]
```diff
@@ -78,16 +78,16 @@ <h3>Sales Totals</h3>
 </div>
 	<form id="topics-filter" method="get">
 		<input
-			type="hidden" name="tribe[page]"
+			type="hidden" name="page"
 			value=""
 		/>
 		<input
-			type="hidden" name="tribe[event_id]"
+			type="hidden" name="post_id"
```
Comment (dpanta94) [inline tests/integration/Tribe/Admin/__snapshots__/OrderReportTest__test_tc_order_report_display__event with 1 pending and 1 completed order__0.snapshot.html:85]: Why is this snapshot being changed ?

File: tests/integration/Tribe/Admin/__snapshots__/OrderReportTest__test_tc_order_report_display__event with multiple tickets and orders__0.snapshot.html [hunk 6]
```diff
@@ -84,16 +84,16 @@ <h3>Sales Totals</h3>
 </div>
 	<form id="topics-filter" method="get">
 		<input
-			type="hidden" name="tribe[page]"
+			type="hidden" name="page"
 			value=""
 		/>
 		<input
-			type="hidden" name="tribe[event_id]"
+			type="hidden" name="post_id"
```
Comment (dpanta94) [inline tests/integration/Tribe/Admin/__snapshots__/OrderReportTest__test_tc_order_report_display__event with multiple tickets and orders__0.snapshot.html:91]: Why is this snapshot being changed ?

**Review by dpanta94 — CHANGES_REQUESTED** [review]
Left some comments. I also noticed phpcs failing, lets make sure we have them passing before merging by either adjusting the Yoda conditions or updating phpcs.xml.

Additionally i noticed a lot of snapshots being updated while the changes don't seem related to them ?

File: src/Tickets/Commerce/Admin/Singular_Order_Page.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (stratease) [reply src/Tickets/Commerce/Admin/Singular_Order_Page.php:112]: We discussed it a while ago, it was removed from other repos but not ET yet. Merging phpcs from TEC.

File: changelog/fix-ET-2184-breadcrumb-single-order-page [hunk 7]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: tweak
+
+Added Orders page breadcrumb for the Order Detail page
```
Comment (stratease) [inline changelog/fix-ET-2184-breadcrumb-single-order-page:4]: This doesn't belong right @dpanta94 ? It is just part of req for the bucket which isn't released right?

File: tests/integration/TEC/Tickets/Emails/Admin/__snapshots__/Emails_TabTest__it_should_match_stored_json_for_general_settings__1.php (hunk 4 above)
Comment (stratease) [reply tests/integration/TEC/Tickets/Emails/Admin/__snapshots__/Emails_TabTest__it_should_match_stored_json_for_general_settings__1.php]: A test was bleeding into others - found and fixed, thanks!

File: changelog/fix-ET-2184-breadcrumb-single-order-page (hunk 7 above)
Comment (dpanta94) [reply changelog/fix-ET-2184-breadcrumb-single-order-page:4]: Yes, if you want for any reason the GH action to be passing, changelogger supports changelog enries of `patch` significance without an `message`. As a result the GH action would pass, but no changelog entry would be added.

Not including the file at all is also fine.

**Comment by stratease** [comment]
@dpanta94 trying to fix [a failing test](https://github.com/the-events-calendar/event-tickets/actions/runs/10494110315/job/29069629980?pr=3179) that seems to be due to no sort being applied. The query tries to apply `order_id` but of course that field is not in the `wp_posts` table. 

Is this supposed to be `ID` or is there a meta field...?

**Comment by dpanta94** [comment]
> @dpanta94 trying to fix [a failing test](https://github.com/the-events-calendar/event-tickets/actions/runs/10494110315/job/29069629980?pr=3179) that seems to be due to no sort being applied. The query tries to apply `order_id` but of course that field is not in the `wp_posts` table.
> 
> Is this supposed to be `ID` or is there a meta field...?

This OrderReportTest is from a previous iteration and if i am not mistaken it uses the ORM. The ORM supports order by order id, since its mapped in the its meta data as a known field.

**Comment by stratease** [comment]
> > @dpanta94 trying to fix [a failing test](https://github.com/the-events-calendar/event-tickets/actions/runs/10494110315/job/29069629980?pr=3179) that seems to be due to no sort being applied. The query tries to apply `order_id` but of course that field is not in the `wp_posts` table.
> > Is this supposed to be `ID` or is there a meta field...?
> 
> This OrderReportTest is from a previous iteration and if i am not mistaken it uses the ORM. The ORM supports order by order id, since its mapped in the its meta data as a known field.

Ok, unfortunately it's not working now. 
image

**Comment by dpanta94** [comment]
> Ok, unfortunately it's not working now.

That's a pity, ill try to take a look as well tomorrow morning. :(

**Comment by stratease** [comment]
> > Ok, unfortunately it's not working now.
> 
> That's a pity, ill try to take a look as well tomorrow morning. :(

In the AttendeeRepository I see it, not the OrderRepository. 

I'll figure it out.

**Review by dpanta94 — APPROVED** [review]
Looks awesome ! :rocket:

**Pull request merged**